### PR TITLE
Add span_names regexp-based filtering to attributes processor

### PR DESCRIPTION
**Pull request by tigrannajaryan**
- Added span_names filter condition.
- Implemented support for regexp patterns in span_names array.
- Added E2E test for attributes processor.
- Added ability to record and verify received data in E2E tests.

**Testing:** 
1. make e2e-test
2. Benchmarked `include` filter using regexp. It takes about 1 microsecond per span for simple regexes, which is virtually the same performance when strict equality is used instead of regex.

**Documentation:** processors README and config examples altered to show the new capability.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@open-telemetry/collector-approvers please review.

File: processor/attributesprocessor/attributes_test.go [hunk 1]
```diff
@@ -867,6 +868,27 @@ func TestAttributes_Matching_False(t *testing.T) {
 				Attributes: []matchAttribute{},
 			},
 		},
+
+		{
+			name: "span name doesn't match",
```
Comment (pjanotti) [inline processor/attributesprocessor/attributes_test.go]: @tigrannajaryan per conversation on other PR, please add the underscores :)

Comment (tigrannajaryan) [reply processor/attributesprocessor/attributes_test.go]: Done.

**Comment by tigrannajaryan** [comment]
@open-telemetry/collector-approvers I need one more review, please.

**Pull request merged**